packages/cozy-scanner/src/Scanner.jsx

```diff
@@ -0,0 +1,259 @@
+                qualification
+              )
+              //It's possible that the filename as changed, so let's update it
+              if (onConflict === 'rename') {
+                this.setState({ filename: newFile.data.name })
+              }
+
+              if (onFinish) onFinish()
+            } catch (error) {
+              this.setState({ error })
+            } finally {
+              this.setState({ status: SCANNER_DONE })
+            }
+          }
+          // Read the file as an ArrayBuffer
+          reader.readAsArrayBuffer(file)
+        },
+        err => {
+          this.setState({ error: err })
+          this.setState({ status: SCANNER_DONE })
+          console.error('error getting fileentry file!' + err)
+        }
+      )
+    }
+
+    const onError = error => {
+      this.setState({ error })
+      this.setState({ status: SCANNER_DONE })
+    }
+    /**
+     * file:/// can not be converted to a fileEntry without the Cordova's File plugin.
+     * `resolveLocalFileSystemURL` is provided by this plugin and can resolve the native
+     * path to a fileEntry readable by a `FileReader`
+     *
+     * When we finished to read the fileEntry as buffer, we start the upload process
+     *
+     */
+
+    window.resolveLocalFileSystemURL(imageURI, onResolvedLocalFS, onError)
+  }
+  //TODO Put this in Files Doctypes
+  /**
+   *
+   * @param {String} name File Name
+   * @param {ArrayBuffer} file data
+   * @param {String} dirId dir id when to upload
+   * @param {String} onConflict Actually only 2 hardcoded strategies 'erase' or 'rename'
+   * @param {Object} metadata An object containing the wanted metadata to attach
+   */
+  async uploadFileWithConflictStrategy(
+    name,
+    file,
+    dirId,
+    onConflict,
+    metadata
+  ) {
+    const { client } = this.props
+    const filesCollection = client.collection('io.cozy.files')
+
+    const path = await CozyFile.getFullpath(dirId, name)
+    try {
+      const existingFile = await filesCollection.statByPath(path)
+      const { id: fileId, dir_id: dirId } = existingFile.data
+      if (onConflict === 'erase') {
+        //!TODO Bug Fix. Seems we have to pass a name attribute ?!
+        const resp = await filesCollection.updateFile(file, {
+          dirId,
+          fileId,
+          name,
+          metadata
+        })
+        return resp
+      } else {
+        const { filename, extension } = CozyFile.splitFilename({
+          name,
+          type: 'file'
+        })
+        const newFileName =
+          CozyFile.generateNewFileNameOnConflict(filename) + extension
+        //recall itself with the newFilename.
+        return this.uploadFileWithConflictStrategy(
+          newFileName,
+          file,
+          dirId,
+          onConflict,
+          metadata
+        )
+      }
+    } catch (error) {
+      if (/Not Found/.test(error)) {
+        return await this.upload(name, file, dirId, metadata)
+      }
+      throw error
+    }
```

Comment:
IMHO the two methods, before being extracted to FileCollection, should be extracted from the component.

Reply:
In fact there are already extracted. I need to use them ;)